src/vulkan-renderer/tools/file.cpp

```diff
@@ -2,44 +2,56 @@
 
 #include <spdlog/spdlog.h>
 
+#include <algorithm>
 #include <cassert>
 #include <fstream>
 
 namespace inexor::vulkan_renderer::tools {
 
-bool File::load_file(const std::string &file_name) {
+bool does_file_exist(const std::string &file_name) {
     assert(!file_name.empty());
 
-    // Open stream at the end of the file to read it's size.
-    std::ifstream file_to_load(file_name.c_str(), std::ios::in | std::ios::binary | std::ios::ate);
-
-    if (file_to_load.is_open()) {
-        spdlog::debug("File {} has been opened.", file_name);
+    std::ifstream file_to_load(file_name.c_str(), std::ios::in);
+    return file_to_load.is_open();
+}
 
-        // Read the size of the file.
-        const auto file_size = file_to_load.tellg();
+std::string get_file_extension_lowercase(const std::string &file_name) {
+    assert(!file_name.empty());
 
-        // Preallocate memory for the file buffer.
-        m_file_data.resize(m_file_size);
+    // Extract the file extension
+    std::string file_extension = file_name.substr(file_name.find_last_of('.') + 1);
 
-        // Reset the file read position to the beginning of the file.
-        file_to_load.seekg(0, std::ios::beg);
+    if (file_extension.empty()) {
```

Comment:
IMHO this function should not raise any exception, as an extension can be empty. Just check "" against the return value.

Reply:
Done

Reply:
Why not use optional?
`std::option<std::string> get_file_extension_lowercase(const std::string &file_name)`

Reply:
We prefer to return an empty string (and empty `std::vector`) over `std::optional` of string.